tests/Doctrine/Tests/ORM/Functional/PaginationTest.php

```diff
@@ -667,6 +667,27 @@ public function testPaginationWithSubSelectOrderByExpression($useOutputWalker, $
         self::assertCount(9, $paginator->getIterator());
     }
 
+    public function testDifferentResultLengthsDoNotRequireExtraQueryCacheEntries(): void
+    {
+        $dql   = 'SELECT u FROM Doctrine\Tests\Models\CMS\CmsUser u WHERE u.id >= :id';
+        $query = $this->_em->createQuery($dql);
+        $query->setMaxResults(10);
+
+        $query->setParameter('id', 1);
+        $paginator = new Paginator($query);
+        $paginator->getIterator(); // exercise the Paginator
+
+        $initialCount = count(self::$queryCache->getValues());
+
+        $query->setParameter('id', 2);
+        $paginator = new Paginator($query);
+        $paginator->getIterator(); // exercise the Paginator again, with a smaller result set
```

Comment:
It might be interesting to be explicit about result set size expectations:

```suggestion
        self::assertCount(9, $paginator->getIterator()); // exercise the Paginator

        $initialCount = count(self::$queryCache->getValues());

        $query->setParameter('id', 2);
        $paginator = new Paginator($query);
        self::assertCount(8, $paginator->getIterator()); // exercise the Paginator again, with a smaller result set
```